File: 백준/Bronze/2596. 비밀편지/비밀편지.py
import sys
input = sys.stdin.readline

# 알파벳 맵 정의
ALPHABET = {
    "000000": "A",
    "001111": "B",
    "010011": "C",
    "011100": "D",
    "100110": "E",
    "101001": "F",
    "110101": "G",
    "111010": "H",
}

# 입력 처리
char_count = int(input().strip())
chars = input().strip()

answer = ""
for i in range(char_count):
    # 6자리씩 잘라서 처리
    target = chars[i * 6:(i + 1) * 6]
    keys = ALPHABET.keys()
    flag = False

    # 각 키와 비교
    for key in keys:
        cnt = sum(1 for j in range(6) if key[j] != target[j])  # 다른 문자 개수 계산
        if cnt <= 1:
            answer += ALPHABET[key]
            flag = True
            break

    # 유효하지 않은 경우
    if not flag:
        answer = str(i + 1)
        break

# 결과 출력
print(answer)

File: 백준/Bronze/2596. 비밀편지/비밀편지.js
const input = require("fs").readFileSync("/dev/stdin").toString().trim().split("\n")


const ALPHABET = {
  "000000": "A",
  "001111": "B",
  "010011": "C",
  "011100": "D",
  100110: "E",
  101001: "F",
  110101: "G",
  111010: "H",
}
const charCount = Number(input[0])
const chars = input[1]
let answer = ""
for (let i = 0; i < charCount; i += 1) {
  const target = chars.slice(i * 6, i * 6 + 6)
  const keys = Object.keys(ALPHABET)
  let flag = false
  for (let key of keys) {
    let cnt = 0
    for (let j = 0; j < 6; j++) {
      if (key[j] !== target[j]) {
        cnt += 1
      }
    }
    if (cnt <= 1) {
      answer += ALPHABET[key]
      flag = true
      break
    }
  }
  if (!flag) {
    answer = (i + 1).toString()
    break
  }
}

console.log(answer)